sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/udf/CreateSpark2ContinuationsFromChangeFeedOffset.scala

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark.udf
+
+import com.azure.cosmos.{CosmosAsyncClient, SparkBridgeInternal}
+import com.azure.cosmos.implementation.SparkBridgeImplementationInternal
+import com.azure.cosmos.implementation.SparkBridgeImplementationInternal.rangeToNormalizedRange
+import com.azure.cosmos.implementation.changefeed.common.ChangeFeedState
+import com.azure.cosmos.implementation.query.CompositeContinuationToken
+import com.azure.cosmos.spark.{ChangeFeedOffset, CosmosClientCache, CosmosClientCacheItem, CosmosClientConfiguration, CosmosConfig, CosmosContainerConfig, Loan}
+import org.apache.spark.sql.api.java.UDF2
+
+import scala.collection.mutable
+
+@SerialVersionUID(1L)
+class CreateSpark2ContinuationsFromChangeFeedOffset extends UDF2[Map[String, String], String, Map[Int, Long]] {
```

Comment:
Spark2 ?
I didn't get the naming?

Reply:
Spark 2 connnector used a different way of storing changefeed continuations in hdfs. This udf allows ADF to in theory migarte a customer from their Spark 2 to Spark 3 environment and also back in case the customer is hitting some regression.